From 9e01ebb2d4c26a18c092285cadb07661fc8db014 Mon Sep 17 00:00:00 2001
From: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
Date: Thu, 14 Dec 2017 13:12:55 +0530
Subject: [PATCH 0991/1240] Soundwire: Fix out of bounds access to
 sdw_num_col_mapping

In sdw_get_col_to_num() the sdw_num_col_mapping can be accessed at
MAX_NUM_COLS index and will lead to out of bound access.

Fix out of bound access by correcting the for loop condition

Change-Id: Ib5f4d392152d674652e64a5732347ad3997d5514
Review-by: Vaibhav Agarwal <vaibhav.agarwal@intel.com>
Signed-off-by: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
---
 drivers/sdw/sdw_bwcalc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/sdw/sdw_bwcalc.c b/drivers/sdw/sdw_bwcalc.c
index 90ae861..57dc85d 100644
--- a/drivers/sdw/sdw_bwcalc.c
+++ b/drivers/sdw/sdw_bwcalc.c
@@ -134,7 +134,7 @@ int sdw_get_col_to_num(int col)
 {
 	int i;
 
-	for (i = 0; i <= MAX_NUM_COLS; i++) {
+	for (i = 0; i < MAX_NUM_COLS; i++) {
 		if (sdw_num_col_mapping[i].col == col)
 			return sdw_num_col_mapping[i].num;
 	}
-- 
2.7.4

